etna/transforms/math/differencing.py

```diff
@@ -96,6 +96,8 @@ def fit(self, df: pd.DataFrame) -> "_SingleDifferencingTransform":
 
             self._train_init_dict[current_segment] = cur_series[: self.period]
         self._test_init_df = fit_df.iloc[-self.period :, :]
+        # make multiindex levels consistent
+        self._test_init_df.columns = self._test_init_df.columns.remove_unused_levels()
```

Comment:
Can we not use `remove_unused_levels`?

Is it the root cause?

Reply:
It is a solution to the problem. After creation of `fit_df` values for old columns remains in `MultiIndex` and it leads to errors during `set_names`.